modules/phase_field/src/auxkernels/calcVelocityAux.C

```diff
@@ -0,0 +1,64 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "calcVelocityAux.h"
+//#include "FEProblem.h"
+
+registerMooseObject("PhaseFieldApp", calcVelocityAux);
+
+template <>
+InputParameters
+validParams<calcVelocityAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  params.addClassDescription(
+      "calculate the velocity of grains while avoiding numerical instability.");
```

Comment:
The "while avoiding numerical instability" is not a particularly useful bit of information and this is rather computing the velocity of *grain boundaries*.